providers/implementations/signature/rsa.c

```diff
@@ -336,9 +336,20 @@ static int rsa_signature_init(void *vprsactx, void *vrsa, int operation)
                     return 0;
                 }
 
-                strncpy(prsactx->mdname, mdname, sizeof(prsactx->mdname));
-                strncpy(prsactx->mgf1_mdname, mgf1mdname,
-                        sizeof(prsactx->mgf1_mdname));
+                if (OPENSSL_strlcpy(prsactx->mdname, mdname,
+                                    sizeof(prsactx->mdname))
+                        >= sizeof(prsactx->mdname)) {
```

Comment:
I am unsure about how to split the line in this `if`, is the current solution acceptable or do you have a better recommendation?

Reply:
Hmm. I ususally tend to indent only lines beginning with || and && by another 4 spaces,
otherwise this looks like
if (a
    op b)
so if a is this rather long strcpy I would not indent the >= by 4 spaces.
But it is sometimes better not to do too much in the if-expression.
That is a general remark, there are much more complicated expressions,
which have no reason for being in an if construct

Reply:
one semantic difference between strncpy and our OPENSSL_strlcpy is,
that strncpy does not need a zero-termination in the input, while strlcpy
does need it (but only for the return value)

Reply:
I went ahead and reformatted it using an auxiliary `size_t` variable to store the result of `OPENSSL_strlcpy()` which then I compare with the size of the buffer.

Seems cleaner now, what do you think?

Reply:
Yeah, I like it better too.